notebooks/official/generative_ai/text_embedding_api_semantic_search_with_scann.ipynb

```diff
@@ -0,0 +1,1153 @@
+{
```

Comment:
<p>Is this an OSS Index or Matching Engine Index?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Reply:
Neither. The code is using the ScaNN library to build a custom index. ScaNN is a library for approximate nearest neighbor search. It provides a variety of algorithms for building and searching indexes, including the tree-based approach used in the code you provided.